TypeScript.YAML-tmLanguage

```diff
@@ -2536,11 +2537,18 @@ repository:
       match: '{{startOfIdentifier}}keyof{{endOfIdentifier}}'
     - name: keyword.operator.ternary.ts
       match: (\?|\:)
-    - name: keyword.operator.expression.infer.ts
-      match: '{{startOfIdentifier}}infer(?=\s+[_$[:alpha:]])'
     - name: keyword.operator.expression.import.ts
       match: '{{startOfIdentifier}}import(?=\s*\()'
 
+  type-infer:
+    patterns:
+    - match: '{{startOfIdentifier}}(infer)\s+({{identifier}})\s+(extends){{endOfIdentifier}}'
```

Comment:
```suggestion
    - match: '{{startOfIdentifier}}(infer)\s+({{identifier}})\s+(?:(extends){{endOfIdentifier}})?'
```

I think you want this to make sure infer T is recognized as well

Reply:
I wonder if it would be easier to treat `extends` as just another type operator, and remove it from `type-conditional` as well.